File: backend/JobTracker.Domain/Entities/Interview.cs
namespace JobTracker.Domain.Entities;

public class Interview
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid JobApplicationId { get; set; }
    public JobApplication? JobApplication { get; set; }
    public string Title { get; set; } = string.Empty;
    public InterviewType Type { get; set; }
    public DateTime ScheduledAt { get; set; }
    public int DurationMinutes { get; set; } = 60;
    public string? InterviewerName { get; set; }
    public string? MeetingLink { get; set; }
    public string? Location { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/interview.ts
export type InterviewType =
  | 0 // PhoneScreen
  | 1 // Technical
  | 2 // Behavioral
  | 3 // FinalRound
  | 4 // Onsite
  | 5; // Other

export type Interview = {
  id: string;
  jobApplicationId: string;
  title: string;
  type: InterviewType;
  scheduledAt: string;
  durationMinutes: number;
  interviewerName: string | null;
  meetingLink: string | null;
  location: string | null;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
};

export type CreateInterviewRequest = {
  title: string;
  type: InterviewType;
  scheduledAt: string;
  durationMinutes: number;
  interviewerName?: string | null;
  meetingLink?: string | null;
  location?: string | null;
  notes?: string | null;
};
